kfs-core/src/main/java/edu/arizona/kfs/fp/batch/service/impl/BankParametersAccessServiceImpl.java

```diff
@@ -0,0 +1,759 @@
+            bofaCustRefNumLength = Integer.parseInt(bofaCustRefNumLengthParm);
+        }
+        return bofaCustRefNumLength;
+    }
+
+    /**
+     * @see edu.arizona.kfs.fp.batch.service.BankParametersAccessService#getBofaCustRefNumRegexFormat()
+     */
+    public String getBofaCustRefNumRegexFormat() {
+        if (bofaCustRefNumRegexFormat == null) {
+            bofaCustRefNumRegexFormat = parameterService.getParameterValueAsString(DocumentCreationStep.class, KFSConstants.BankTransactionsParameters.BOFA_CUST_REF_NUM_REGEX_FORMAT);
+        }
+        return bofaCustRefNumRegexFormat;
+    }
+
+    /**
+     * @see edu.arizona.kfs.fp.batch.service.BankParametersAccessService#getBofaCustRefNumGroupIndex()
+     */
+    public Integer getBofaCustRefNumGroupIndex() {
+        if (bofaCustRefNumGroupIndex == null) {
+            String bofaCustRefNumGroupIndexParm = parameterService.getParameterValueAsString(DocumentCreationStep.class, KFSConstants.BankTransactionsParameters.BOFA_CUST_REF_NUM_GROUP_INDEX);
+            bofaCustRefNumGroupIndex = Integer.parseInt(bofaCustRefNumGroupIndexParm);
+        }
+        return bofaCustRefNumGroupIndex;
+    }
+
+    /**
+     * @see edu.arizona.kfs.fp.batch.service.BankParametersAccessService#getBofaNotesRegexFormat()
+     */
+    public String getBofaNotesRegexFormat() {
+        if (bofaNotesRegexFormat == null) {
+            bofaNotesRegexFormat = parameterService.getParameterValueAsString(DocumentCreationStep.class, KFSConstants.BankTransactionsParameters.BOFA_NOTES_REGEX_FORMAT);
+        }
+        return bofaNotesRegexFormat;
+    }
+
+    /**
+     * @see edu.arizona.kfs.fp.batch.service.BankParametersAccessService#getBofaNotesIndicator()
+     */
+    public String getBofaNotesIndicator() {
+        if (bofaNotesIndicator == null) {
+            bofaNotesRegexFormat = parameterService.getParameterValueAsString(DocumentCreationStep.class, KFSConstants.BankTransactionsParameters.BOFA_NOTES_INDICATOR);
+        }
+        return bofaNotesIndicator;
+    }
+
+
+    /**
+     * @see edu.arizona.kfs.fp.batch.service.BankParametersAccessService#getBofaCcAccountNumbers()
+     */
+    public Set<String> getBofaCcAccountNumbers() {
+        if (bofaCcAccountNumbers == null) {
+            bofaCcAccountNumbers = getValueSetForParameter(KFSConstants.BankTransactionsParameters.BOFA_CC_ACCOUNT_NUM_SET, getBofaDelimiter());
+        }
+        return bofaCcAccountNumbers;
+    }
+
+
+    /**
+     * @see edu.arizona.kfs.fp.batch.service.BankParametersAccessService#getBofaCcBaiTypeCodes()
+     */
+    public Set<String> getBofaCcBaiTypeCodes() {
+        if (bofaCcBaiTypeCodes == null) {
+            bofaCcBaiTypeCodes = getValueSetForParameter(KFSConstants.BankTransactionsParameters.BOFA_CC_BAI_TYPE_CODE_SET, getBofaDelimiter());
+        }
+        return bofaCcBaiTypeCodes;
+    }
+
+
+    /******************* VA specific parameters ************************/
```

Comment:
Silly, but the double kleene star will trigger the javadoc to try and generate docs off this line.